### PR TITLE
[MNG-6863] - Fixing that --also-make is being ignored when calling --resume-from

### PR DESCRIPTION
**Pull request by MartinKanters**
Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MNG-6863) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[MNG-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MNG-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [x] You have run the [Core IT][core-its] successfully.

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [x] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

[core-its]: https://maven.apache.org/core-its/core-it-suite/

### CODE REVIEW

File: maven-core/src/test/java/org/apache/maven/graph/DefaultGraphBuilderTest.java [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+                scenario( "Selected project with also make dependents" )
+                        .selectedProjects( singletonList( MODULE_B ) )
+                        .makeBehavior( REACTOR_MAKE_DOWNSTREAM )
+                        .expectResult( asList( MODULE_B, MODULE_C ) ),
+                scenario( "Resuming from project with also make dependencies" )
+                        .makeBehavior( REACTOR_MAKE_UPSTREAM )
+                        .resumeFrom( MODULE_C )
+                        .expectResult( asList( MODULE_A, MODULE_B, MODULE_C ) ),
+                scenario( "Selected project with resume from an also make dependency (MNG-4960 IT#1)" )
+                        .selectedProjects( singletonList( MODULE_C ) )
```
Comment (rfscholte) [inline maven-core/src/test/java/org/apache/maven/graph/DefaultGraphBuilderTest.java:117]: This selectedProjects is awkward: it says `resume from B`, which already implies running C.
In general I think that `resume-from` should ignore the project-list (until https://issues.apache.org/jira/browse/MNG-6676 is added, which means you can exclude modules from the result-from)

File: maven-core/src/test/java/org/apache/maven/graph/DefaultGraphBuilderTest.java [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+                        .makeBehavior( REACTOR_MAKE_UPSTREAM )
+                        .expectResult( asList( MODULE_A, MODULE_C ) ),
+                scenario( "Resume from exclude project downstream" )
+                        .resumeFrom( MODULE_A )
+                        .excludedProjects( singletonList( MODULE_B ) )
+                        .expectResult( asList( MODULE_A, MODULE_C ) ),
+                scenario( "Exclude the project we are resuming from (as proposed in MNG-6676)" )
+                        .resumeFrom( MODULE_B )
+                        .excludedProjects( singletonList( MODULE_B ) )
+                        .expectResult( singletonList( MODULE_C ) )
```
Comment (rfscholte) [inline maven-core/src/test/java/org/apache/maven/graph/DefaultGraphBuilderTest.java:143]: Ah, this is MNG-6676. Nice

**Comment by rfscholte** [comment]
I really like these tests: easy to read and to verify all kinds of combinations. Well done!

**Comment by MartinKanters** [comment]
> I really like these tests: easy to read and to verify all kinds of combinations. Well done!

Thanks! I was doubting this setup since it seems a bit unusual in the maven codebase (afaik), but I'm happy that it helps with the readability!

File: maven-core/src/test/java/org/apache/maven/graph/DefaultGraphBuilderTest.java (hunk 1 above)
Comment (MartinKanters) [reply maven-core/src/test/java/org/apache/maven/graph/DefaultGraphBuilderTest.java:117]: Yes, I agree it's a weird scenario, but we have an integration test which verifies exactly this. (MNG-4960)

File: maven-core/src/test/java/org/apache/maven/graph/DefaultGraphBuilderTest.java (hunk 2 above)
Comment (MartinKanters) [reply maven-core/src/test/java/org/apache/maven/graph/DefaultGraphBuilderTest.java:143]: I was wondering whether you would like MNG-6676, but I've got my answer now, I guess :)

File: maven-core/src/test/java/org/apache/maven/graph/DefaultGraphBuilderTest.java (hunk 1 above)
Comment (rfscholte) [reply maven-core/src/test/java/org/apache/maven/graph/DefaultGraphBuilderTest.java:117]: Confirmed

**Comment by jira-importer** [comment]
Resolve #7826

**Pull request merged**